build-recipe-openwrt

```diff
@@ -0,0 +1,128 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+################################################################
+
+
+recipe_setup_openwrt() {
+    echo $BUILD_ROOT
+    pwd
+    TOPDIR=/usr/src/packages
+    "$DO_INIT_TOPDIR" && rm -rf "$BUILD_ROOT$TOPDIR"
+    for i in OTHER SOURCES ; do
+	mkdir -p "$BUILD_ROOT$TOPDIR/$i"
+    done
+    if test "$MYSRCDIR" = $BUILD_ROOT/.build-srcdir ; then
+	mv "$MYSRCDIR"/* $BUILD_ROOT$TOPDIR/SOURCES/
+    else
+	copy_sources "$MYSRCDIR" "$BUILD_ROOT$TOPDIR/SOURCES/"
+    fi
+    chown -hR "$ABUILD_UID:$ABUILD_GID" "$BUILD_ROOT$TOPDIR"
+}
+
+recipe_prepare_openwrt() {
+    :
+}
+
+# Variables:
+# $BUILD_ROOT is the chroot
+# $TOPDIR/SOURCES includes the sources
+# $RECIPEFILE the name of the openwrt.(yaml|json) config file
+
+recipe_build_openwrt() {
+    echo "===================== recipe_build_openwrt ($@)"
+    set -e
+
+    ls -l $BUILD_ROOT$TOPDIR/SOURCES
+    cd "$BUILD_ROOT$TOPDIR/"
+    cp $BUILD_ROOT$TOPDIR/SOURCES/* .
+
+    local OPENWRT_FILE="$TOPDIR/SOURCES/$RECIPEFILE"
+    export OPENWRT_FILE
+    export TOPDIR
+    export RELEASE
+
+
+    local NAME=$(perl -I$BUILD_DIR -MBuild::OpenWrt -e 'Build::OpenWrt::show' -- "$BUILD_ROOT$OPENWRT_FILE" name)
+    local GITURL=$(perl -I$BUILD_DIR -MBuild::OpenWrt -e 'Build::OpenWrt::show' -- "$BUILD_ROOT$OPENWRT_FILE" giturl)
+    local SDKURL=$(perl -I$BUILD_DIR -MBuild::OpenWrt -e 'Build::OpenWrt::show' -- "$BUILD_ROOT$OPENWRT_FILE" sdkurl)
+    local ARCH=$(perl -I$BUILD_DIR -MBuild::OpenWrt -e 'Build::OpenWrt::show' -- "$BUILD_ROOT$OPENWRT_FILE" arch)
+    local SUBARCH=$(perl -I$BUILD_DIR -MBuild::OpenWrt -e 'Build::OpenWrt::show' -- "$BUILD_ROOT$OPENWRT_FILE" subarch)
+    local RELVER=$(perl -I$BUILD_DIR -MBuild::OpenWrt -e 'Build::OpenWrt::show' -- "$BUILD_ROOT$OPENWRT_FILE" relver)
+    local TARBALL=$(perl -I$BUILD_DIR -MBuild::OpenWrt -e 'Build::OpenWrt::show' -- "$BUILD_ROOT$OPENWRT_FILE" tarball)
+    
+    export NAME
+    export GITURL
+    export ARCH
+    export SUBARCH
+    export RELVER
+    export SDKURL
+    export TARBALL
+
+    tar xf "$TARBALL" -C "$BUILD_ROOT/repo/"
+    local URL=$(grep PKG_SOURCE_URL: "$BUILD_ROOT/repo/$NAME/Makefile" | cut -f2 -d '=')
+    if [ "$URL" != '/git' ]
+    then
+	rm -rf "$BUILD_ROOT/git"
+        git clone "$URL" "$BUILD_ROOT/git"
+        sed -i "s|$URL|/git|" "$BUILD_ROOT/repo/$NAME/Makefile"
+    fi
+
+
+    chroot "$BUILD_ROOT" bash -c "cd sdk && grep localrepo feeds.conf.default || echo \"src-link localrepo /repo/\" >> feeds.conf.default"
+
+
+    # Download the dependencies
+    chroot "$BUILD_ROOT" /bin/bash -c "cd sdk && make download V=s"
```

Comment:
This will fail on OBS where networking is disabled.

Reply:
I am still wondering what I need to do about build-time dependencies (`make package/$NAME/download`), do I assume those are passed in as build inputs somehow?

Reply:
I'm currently thinking of maybe having a list of git repos that are copied in that are comitted in the source build tree, though I'm not sure how that interacts with package/$NAME/download.

Reply:
I have a question -- do you know if I'm able to use https://github.com/openSUSE/obs-build/blob/master/PBuild/RemoteAssets.pm before building?

Reply:
depends, pbuild is executing it always. 

For OBS scenarios the "download_asset" service would need to get enabled for that package or project.

Reply:
regarding dependencies, you may define a default as     build-packages:openwrt     in Build.pm in %subst_defaults.

This would require these packages by default (but can be overwritten by the user). 

You may also define an openwrt config file where you define default repository/ies via RepoURL: tag